diff --git a/dist/sharp.cjs b/dist/sharp.cjs
index 8ee430134916138faf843db0c1097a29053def5e..dfa763beaecfdfc8a0e77853ef65daf767b09a13 100644
--- a/dist/sharp.cjs
+++ b/dist/sharp.cjs
@@ -21,8 +21,12 @@ const runtimePlatform = runtimePlatformArch();
 
 let sharp;
 const errors = [];
+// Electron's Linux GLib conflicts with the native libvips bundled by Sharp.
+if (process.platform === 'linux' && process.versions.electron) {
+  sharp = require('@img/sharp-wasm32/sharp.node');
+}
 try {
-  sharp = require(`../src/build/Release/sharp-${runtimePlatform}-${version}.node`);
+  if (!sharp) sharp = require(`../src/build/Release/sharp-${runtimePlatform}-${version}.node`);
 } catch (err) {
   errors.push(err);
 }
diff --git a/dist/sharp.mjs b/dist/sharp.mjs
index d1b6bd4a5f758b535435edca7318813009190ec5..ca6d071c19473221e8d670dc4a7199479a17531b 100644
--- a/dist/sharp.mjs
+++ b/dist/sharp.mjs
@@ -21,8 +21,12 @@ const runtimePlatform = runtimePlatformArch();
 
 let sharp;
 const errors = [];
+// Electron's Linux GLib conflicts with the native libvips bundled by Sharp.
+if (process.platform === 'linux' && process.versions.electron) {
+  sharp = require('@img/sharp-wasm32/sharp.node');
+}
 try {
-  sharp = require(`../src/build/Release/sharp-${runtimePlatform}-${version}.node`);
+  if (!sharp) sharp = require(`../src/build/Release/sharp-${runtimePlatform}-${version}.node`);
 } catch (err) {
   errors.push(err);
 }
